### ISSUE TITLE
Towards Open World Recognition

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/readinglist#361]
https://arxiv.org/abs/1412.5687

**Comment by sweaterr** [comment]
Distance based classifiers like Nearest Class Mean (NCM) [17, 31, 36] offer a natural choice for building scalable system that can learn new classess incrementally.

In NCM-like classiﬁers, incorporating new images or classes in implies adjusting the existing means or updating the set of class means.

However, NCM classiﬁer in its current formulation is not suited for open set recognition because it uses close-set assumptions for probability normalization.

Handling unknowns in open world recognition requires gradual decrease in the value of probability (of class membership) as the test point moves away from known data into open space.

The Softmax based probability assignment used in NCM does not account for open space.

**Comment by sweaterr** [comment]
The ﬁrst contribution of this paper is a formal deﬁnition of the problem of open world recognition, which extends the existing deﬁnition of open set recognition which was deﬁned for a static notion of set.

In order solve open world recognition, the system needs be robust to unknown classes, but also be able to move through the stages and knowledge progression summarized in Fig  1.

Second contribution of the work is a recognition system that can continuously learn new object categories in an open world model.

In particular, we show how to extend Nearest Class Mean type algorithms (NCM) [31], [36], to a Nearest Non-Outlier (NNO) algorithm that can balance open space risk and accuracy.

**Comment by sweaterr** [comment]
To support this extension, our third contribution is showing that thresholding sums of monotonically decreasing functions of distances of linearly transformed feature space can have arbitrarily small “open space risk”.

Finally, we present a protocol for evaluation for open world recognition, and use this protocol to show our NNO algorithm perform signiﬁcantly better on open world recognition evaluation using Image-Net [2].

**Comment by sweaterr** [comment]
# 2 Related Work

Our work addresses an issue that is related to and has received attention from various communities such as incremental learning, scalable and open set learning.

**Comment by sweaterr** [comment]
## Incremental Learning

As SVMs rose to prominence in  many object recognition [52, 25], many incremental extensions to SVMs were proposed.

Cauwenberghs et al, [4] proposed an incremental binary SVMs by means of saving and updating KKT conditions.

Yeh et al, [51] extended the approach to object recognition and demonstrated multi-class incremental learning.

Pronobis [35] proposed memorycontrolled online incremental SVM for visual place recognition.

Although incremental SVMs might seem like a natural for large scale incremental learning for object recognition, they suffer from multiple drawbacks.

The update process is extremely expensive (quadratic in the number of training examples learned [21]) and depends heavily on the number of support vectors stored for performing updates [21].

To overcome the update expense, [5] and [41] proposed classiﬁers with fast and inexpensive update process along with their multi-class extensions.

However, the multiclass incremental learning methods and other incremental classiﬁers, [5, 41, 49, 24], are incremental in terms of additional training samples.

**Comment by sweaterr** [comment]
## Scalable Learning: 

Researchers like [30, 26, 11] have proposed label tree based classiﬁcation methods to address scalability (# of object categories) in large scale visual recognition challenges [45, 2]. 

Recent advances in deep learning community [18], [43] has resulted in state of the art performance on these challenges. Such methods are extremely useful when the goal is obtain maximum classiﬁcation/recognition performance. These systems assume a priori availability of entire training data (images and categories). However, adapting such methods to a dynamic learning scenario becomes extremely challenging. Adding object categories requires retraining the entire system, which could be infeasible for many applications. Thus, these methods are scalable but not incremental (Fig 2)

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39125964-c0ee8aae-473b-11e8-85b2-1d33b9f7660a.png)

**Comment by sweaterr** [comment]
The ﬁnal aspect of related work is nearest class mean (NCM) classiﬁers.

NCM classiﬁcation, in which samples undergo a Manhanalobis transform and then are are associated with a class/cluster mean, is a classic pattern recognition approach [14].

NCM classiﬁers have a long history of use in vision systems, [7] and have multiple extensions, adapations and applications [9, 44, 50, 20, 27].

Recently the technique has been adapted for use in larger scale vision problems [48, 47, 31, 36], with the most recent and most accurate approaches combining NCM with metric learning [31] and with random forests[36].

**Comment by sweaterr** [comment]
# 3 Open World Recognition

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39126708-fafda462-473d-11e8-949b-f66a1b9fc216.png)

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39127189-3de0143a-473f-11e8-93f4-7f177d429cc4.png)

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39127237-5f4c78e8-473f-11e8-87ba-18dbba68b3ce.png)

**Comment by sweaterr** [comment]
Testing Phase:
* known class 500 unknown class 500 으로 나눈다.

**Comment by sweaterr** [comment]
## 5.2 Experimental Results
* 50 개씩 클래스를 추가해가면서 학습

![image](https://user-images.githubusercontent.com/1518919/39128134-f5462c0c-4741-11e8-99e8-20fbd1bf5e39.png)
